docs/scope-and-objectives.md

```diff
@@ -0,0 +1,139 @@
+- To provide a transition path for Kubernetes lifecycle products to adopt Cluster API incrementally. The result is a fragmentation of tooling, and we want to create a unified mechanism that gives users consistency around cluster management.
+
+### Non-goals
+
+- To add these APIs to Kubernetes core (kubernetes/kubernetes).
+   -  This API should live in a namespace outside the core and follow the best practices defined by api-reviewers, but is not subject to core-api constraints.
+- To manage the lifecycle of infrastructure unrelated to the running of Kubernetes-conformant clusters.
+- To force all Kubernetes lifecycle products (kops, kubespray, GKE, AKS, EKS, IKS etc.) to support or use these APIs.
+- To manage non-Cluster API provisioned Kubernetes-conformant clusters.
+- To manage a single cluster spanning multiple infrastructure providers.
+- To configure a machine at any time other than create or upgrade.
+- To duplicate functionality that exists or is coming to other tooling, e.g., updating kubelet configuration (c.f. dynamic kubelet configuration), or updating apiserver, controller-manager, scheduler configuration (c.f. component-config effort) after the cluster is deployed.
+
+## Requirements
+
+#### Foundation
+
+- Cluster API MUST be able to be deployed and run on Kubernetes.
+
+- Cluster API MUST provide isolation between providers.
+
+- Cluster API MUST support multiple infrastructure providers, including both on-prem and cloud providers.
+
+- Cluster API MUST be able to manage groups/sets of Kubernetes nodes that support scaling and orchestrated upgrades.
+
+- Cluster API, through a single installation, MUST be able to manage (create, scale, upgrade, destroy) clusters on multiple providers.
+
+- Cluster API SHOULD support all operating systems in scope for Kubernetes conformance.
+
+#### User Experience
+
+- Cluster API MUST be able to display the versions of Kubernetes and related components of a Node that it manages.
+
+- Cluster API MUST be able to provide sufficient information for a consumer to access a provisioned Kubernetes cluster’s API server.
+
+#### Organization
+
+- Cluster API MUST NOT have code specific to a particular provider within the main repository, defined as sigs.k8s.io/cluster-api.
+
+#### Validation
+
+- Cluster API provider implementations SHOULD validate configuration data.
+
+- Cluster API MUST test integrations that it claims will work.
+
+- Cluster API MUST deploy clusters that pass Kubernetes conformance tests.
+
+- Cluster API MUST document how to bootstrap itself and MAY provide tooling to do so.
+
+- Cluster API MUST document requirements for machine images and SHOULD reference default tools to prepare machine images.
+
+- Cluster API MUST have conformance tests to ensure that Cluster API and a given provider fulfill the expectations of users/automation.
+
+#### Extension
```

Comment:
As we look to centralize a lot of the logic that is being handled by providers today, I kind of envision it being a separate type of provider. Of course, that is all yet to be determined as part of the design. However, while we should provide default behavior for a lot of the common logic, I wouldn't call it core functionality because I fully expect there will be other implementations over time.

Reply:
@detiber  Agree with your comment but I don't see how this address my concern about naming this section "Extension" given the fact that only the first item is actually about extensibility while the rest is a list of functionalities.

Reply:
I believe @vincepri has everything that is an extension point grouped under this Extension heading. That is why it is organized this way.

Reply:
We don't necessarily have to keep it this way, however.

Reply:
As @ncdc mentioned, I reorganized the items that can be extended under this section to make it clear that they follow the same pattern and requirements that's expressed in the first sentence

Reply:
@pablochacin I'm going to resolve this for now, we can always change it up later!